receiver/postgresqlreceiver/config.go

```diff
@@ -0,0 +1,77 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgresqlreceiver
+
+import (
+	"errors"
+	"fmt"
+	"net"
+
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+	"go.uber.org/multierr"
+)
+
+// Errors for missing required config parameters.
+const (
+	ErrNoUsername          = "invalid config: missing username"
+	ErrNoPassword          = "invalid config: missing password" // #nosec G101 - not hardcoded credentials
+	ErrNotSupported        = "invalid config: field '%s' not supported"
+	ErrTransportsSupported = "invalid config: 'transport' must be 'tcp' or 'unix'"
+	ErrHostPort            = "invalid config: 'endpoint' must be in the form <host>:<port> no matter what 'transport' is configured"
+)
+
+type Config struct {
+	scraperhelper.ScraperControllerSettings `mapstructure:",squash"`
+	Username                                string                   `mapstructure:"username"`
+	Password                                string                   `mapstructure:"password"`
+	Databases                               []string                 `mapstructure:"databases"`
+	confignet.NetAddr                       `mapstructure:",squash"` // provides Endpoint and Transport
+	configtls.TLSClientSetting              `mapstructure:",squash"` // provides Endpoint and Transport
```

Comment:
don't squash, put everything under "tls:" to be consistent with https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L41

Reply:
Done.